src/test/scala/org/neo4j/spark/SparkConnectorScalaBaseTSE.scala

```diff
@@ -0,0 +1,49 @@
+package org.neo4j.spark
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.junit._
+import org.neo4j.driver.{AuthTokens, Driver, GraphDatabase, Session}
+
+
+object SparkConnectorScalaBaseTSE {
+
+  private var startedFromSuite = true
+
+  @BeforeClass
+  def setUpContainer() = {
+    if (!SparkConnectorScalaSuiteIT.server.isRunning) {
+      startedFromSuite = false
+      SparkConnectorScalaSuiteIT.setUpContainer()
+    }
+  }
+
+  @AfterClass
+  def tearDownContainer() = {
+    if (!startedFromSuite) {
+      SparkConnectorScalaSuiteIT.tearDownContainer()
+    }
+  }
+
+}
+
+class SparkConnectorScalaBaseTSE {
+
+  val conf: SparkConf = SparkConnectorScalaSuiteIT.conf
+  val sc: SparkContext = SparkConnectorScalaSuiteIT.sc
+
+  private var driver: Driver = _
+
+  def session(): Session = {
+    if (driver == null) {
```

Comment:
shouldn't we allocate teh driver at the beginning of the test?

Reply:
The container instance is one for all tests so I think we can do it here